src/compiler/tsbuildPublic.ts

```diff
@@ -1272,16 +1295,17 @@ namespace ts {
         program: T | undefined,
         config: ParsedCommandLine,
         diagnostics: readonly Diagnostic[],
-        errorFlags: BuildResultFlags,
-        errorType: string
+        buildResult: BuildResultFlags,
+        errorType: string,
     ) {
+        const canEmitBuildInfo = !(buildResult & BuildResultFlags.SyntaxErrors) && program && !outFile(program.getCompilerOptions());

```

Comment:
Why do syntax errors block a build info?

Reply:
Because there is nothing to write in this case. Only semantic errors are cached so there is no use writing tsbuildinfo in this case